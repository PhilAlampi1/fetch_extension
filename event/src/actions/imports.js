export const setFormMappingData = (sfId, riId) => ({
    type: 'SET_FORM_MAPPING_DATA',
    formMappingStandardFieldId: sfId,
    formMappingRowIdentifierId: riId
})

export const createUpdateUserFormFieldMappingInDb = () => ({
    type: 'CREATE_UPDATE_USER_FORM_FIELD_MAPPING_IN_DB'
})

export const resetFormMappingFields = () => ({
    type: 'RESET_FORM_MAPPING_FIELDS',
    formMappingRowIdentifierId: null,
    formMappingStandardFieldId: null,
    enteredDefaultValue: null,
    enteredDefaultOverride: null,
    rightClickedFormElementType: null,
    rightClickedFormElementValue: null,
    rightClickedFormElementOptions: null,
    defaultValueConfirmed: false,
    defaultOverrideConfirmed: false,
    formFieldSelector: null,
    rightClickSelectionIsValid: false
})