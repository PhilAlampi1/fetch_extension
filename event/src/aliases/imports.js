import {
    status,
    json,
    serverPath,
    formatForXhr,
    findAndStoreImportFileSetups,
    fetchStubValues,
    findAndStoreUserForms,
    setupContextMenu
} from '../utilities/utilities'

export const updateExistingImportFileNameInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'updateimportfilesetupname/' +
            state.imports.selectedImportFileSetupId + '/' +
            state.auth.userToken + '/' +
            state.imports.selectedImportFileSetupName
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'CONFIRM_EXISTING_IMPORT_FILE_NAME',
                    importFileNameConfirmed: true
                })
            })
    }
}

export const updateImportFieldMappingsInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        const importFieldMappings = state.imports.importFieldMappingDbUpdates
        const tok = state.auth.userToken
        const data = formatForXhr([{
            name: 'values',
            value: JSON.stringify(importFieldMappings)
        }, {
            name: 'userToken',
            value: tok
        }])
        const xhr = new XMLHttpRequest()
        xhr.onreadystatechange = () => {
            if (xhr.readyState == XMLHttpRequest.DONE) {
                const res = JSON.parse(xhr.responseText)
                if (res.success) {
                    return dispatch({
                        type: 'COMPLETE_UPDATE_IMPORT_FIELD_MAPPINGS_IN_DB',
                        importFieldMappingDbUpdates: []
                    })
                } else {
                    // TODO - ADD ERROR HANDLING TO REDUCERS AND THROW ERROR HERE
                    // UPDATE UI COMPONENT TO DISPLAY ERROR
                    console.log('DATABASE REQUEST ERROR', res)
                }
            }
        }
        xhr.open("POST", serverPath + 'importfieldmappings/update')
        xhr.setRequestHeader("content-type", "application/x-www-form-urlencoded");
        xhr.setRequestHeader("cache-control", "no-cache");
        xhr.send(data)
    }
}

export const createImportFileSetup = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'createimportfilesetup/' +
            state.imports.selectedImportFileSetupName + '/' +
            state.auth.userId + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                // Insert all Import_Field_Mapping records into DB
                // Create Import_Field_Mapping array out of state
                let importFieldMappings = []
                const standardFieldsCopy = state.imports.standardFields
                standardFieldsCopy.map(sf => {
                    if (sf.importedFieldName) {
                        importFieldMappings.push({
                            userId: state.auth.userId,
                            importFileSetupId: result.data.importFileSetupId,
                            standardFieldId: sf.standardFieldId,
                            importedFieldName: sf.importedFieldName
                        })
                    }
                })
                const tok = state.auth.userToken
                const data = formatForXhr([{
                    name: 'values',
                    value: JSON.stringify(importFieldMappings)
                }, {
                    name: 'userToken',
                    value: tok
                }])
                const xhr = new XMLHttpRequest()
                xhr.onreadystatechange = () => {
                    if (xhr.readyState == XMLHttpRequest.DONE) {
                        const res = JSON.parse(xhr.responseText)
                        if (res.success) {
                            findAndStoreImportFileSetups(state.auth.userId, state.auth.userToken, dispatch)
                            return dispatch({
                                type: 'COMPLETE_CONFIRM_NEW_IMPORT_FILE_NAME',
                                selectedImportFileSetupId: result.data.importFileSetupId,
                                importFileNameConfirmed: true
                            })
                        } else {
                            // TODO - ADD ERROR HANDLING TO REDUCERS AND THROW ERROR HERE
                            // UPDATE UI COMPONENT TO DISPLAY ERROR
                            console.log('DATABASE REQUEST ERROR', res)
                        }
                    }
                }
                xhr.open("POST", serverPath + "importfilemappings/add")
                xhr.setRequestHeader("content-type", "application/x-www-form-urlencoded");
                xhr.setRequestHeader("cache-control", "no-cache");
                xhr.send(data)
            })
    }
}

export const createNewForm = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'createform/' +
            state.imports.selectedFormName + '/' +
            state.imports.selectedFormDescription + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'SET_SELECTED_FORM_ID',
                    selectedFormId: result.data.formId
                })
            })
    }
}

export const getAndStoreImportFieldMappings = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'findimportfieldmappings/' +
            state.imports.selectedImportFileSetupId + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                // Instead of returing ImportFieldMappings, update standardFields for the user's selected Import File Setup
                // and replace the old standardFields with the new standardFields in the store               
                return dispatch({
                    type: 'STORE_IMPORT_FIELD_MAPPINGS',
                    standardFields: result.data
                })
            })
    }
}

export const fetchStubValuesFromAlias = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        const currentPage = state.imports.usersCurrentPage
        fetchStubValues()
            .then(() => {
                findAndStoreImportFileSetups(state.auth.userId, state.auth.userToken, dispatch)
                findAndStoreUserForms(state.auth.userToken, dispatch)
                if (currentPage !== 'main') {
                    return dispatch({
                        type: 'UPDATE_USERS_CURRENT_PAGE',
                        usersCurrentPage: 'main'
                    })
                }
            })
    }
}

export const updateFormInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'updateform/' +
            state.imports.selectedFormId + '/' +
            state.imports.selectedFormName + '/' +
            state.imports.selectedFormDescription + '/' +
            state.imports.selectedFormPublic + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'CONFIRM_UPDATE_FORM',
                    formConfirmed: true
                })
            })
    }
}

export const activateContextMenus = () => {
    return () => setupContextMenu()
}

export const removeContextMenus = () => {
    return () => chrome.contextMenus.removeAll()
}

