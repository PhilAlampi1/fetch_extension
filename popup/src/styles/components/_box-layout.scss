.box-layout {
    background: url('/images/buildings.jpeg');
    background-size: cover;
    display: flex;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(black, 0.25);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    margin: auto;
}

.box-layout__title {
    color: white;
    font-size: $xl-size;
    font-weight: bold;
}

.box-layout__subtext {
    color: white;
    font-size: $m-size;
    font-weight: 500;
    padding: 0 0 $s-size 0;
    text-align: center;
}

.box-layout__footer {
    display: flex;
    justify-content: space-around;
}

.box-layout__footertext {
    color: white;
    font-size: $s-size;
    margin: $s-size;
}

.box-layout__disclaimer {
    margin: $xl-size 0 0 0;
}