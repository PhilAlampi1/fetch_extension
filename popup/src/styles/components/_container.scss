.container {
    margin: 0 auto;
    max-width: 60rem;
    padding: 0 $m-size;
    overflow: scroll;
}

.container__centertext {
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__lefttext {
    text-align: left;
    margin: 0;
    overflow: scroll;
}

.container__flex {
    display: flex;
    justify-content: space-around;
    overflow: scroll;
}

.container--center {
    display: flex;
    align-content: center;
    justify-content: center;
    overflow: scroll;
}

.container__flexvertical {
    display: flex;
    flex-direction: column;
  }

.container__addspace {
    margin: $s-size 0;
}

.container__centerify {
    display: flex;
    align-items: center;
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__child {
    flex: 1;
    overflow: scroll;
}

.add-space {
    margin: $xl-size 0 0 0;
}