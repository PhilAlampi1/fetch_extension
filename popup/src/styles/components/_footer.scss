.footer {
    display: flex;
    justify-content: space-around;
}

.footer-item {
    color: black;
    font-size: $m-size;
    margin: $xl-size $s-size 1rem $s-size;
}