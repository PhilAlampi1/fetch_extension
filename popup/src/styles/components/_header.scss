.header {
    align-items: flex-start;
    background: $off-black;
    color: white;
    display: flex;
    padding: $m-size 0;
}

.header__container {
    align-items: center;
    display: flex;
    flex: 1;
    justify-content: space-between;
    padding: 0 $m-size;
}

.header__title {
    font-size: $l-size;
    margin: 0 $m-size;
    padding: $s-size;
    text-align: left;
    flex: 1;
}

.header__linkgroup {
    flex: 4;
}

.header__links {
    color: $off-white;
    font-size: $m-size;
    font-weight: 500;
    margin: $s-size;
    padding: $s-size;
    text-align: center;
}

.header__span {
    width: 100%
}

.header-question-parent {
    align-items: center;
    display: flex;
    justify-content: flex-start;
}

.header-question-child {
    font-size: $m-size;
    margin: 0 $s-size;
    padding: $s-size;
    width: 100px;
}