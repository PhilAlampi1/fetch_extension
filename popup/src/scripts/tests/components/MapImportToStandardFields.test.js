import React from 'react'
import { shallow } from 'enzyme'
import { MapImportToStandardFields } from '../../components/MapImportToStandardFields'
import { standardFields } from '../../../../../event/src/tests/fixtures/imports'

const onClickSpyOk = jest.fn()
const onClickSpyError = jest.fn()
const standardFieldsNoImportRowIdentifier = [{
    standardFieldId: "1",
    standardFieldName: "Mapping Field (IDX)",
    standardFieldDescription: "A field for mapping.",
    importRowIdentifier: false,
    importedFieldName: "IDX Address Mapg"
}, {
    standardFieldId: "2",
    standardFieldName: "Type of Property",
    standardFieldDescription: "The type of building or structure the property is.",
    importRowIdentifier: false,
    importedFieldName: "Property Type"
}, {
    standardFieldId: "3",
    standardFieldName: "Address",
    standardFieldDescription: "Address of property.",
    importRowIdentifier: false, // changed this from 'true' for test
    importedFieldName: "Street Address"
}]
const standardFieldsNoImportedFieldName= [{
    standardFieldId: "1",
    standardFieldName: "Mapping Field (IDX)",
    standardFieldDescription: "A field for mapping.",
    importRowIdentifier: false,
    importedFieldName: "IDX Address Mapg"
}, {
    standardFieldId: "2",
    standardFieldName: "Type of Property",
    standardFieldDescription: "The type of building or structure the property is.",
    importRowIdentifier: false,
    importedFieldName: "Property Type"
}, {
    standardFieldId: "3",
    standardFieldName: "Address",
    standardFieldDescription: "Address of property.",
    importRowIdentifier: true,
    importedFieldName: "" // changed this from "Street Address" for test
}]

test('MapImportToStandardFields renders correctly', () => {
    const wrapper = shallow(<MapImportToStandardFields
        standardFields={standardFields}
        updateImportFileMapping={onClickSpyOk}
        setMapImportToStandardFieldsERROR={onClickSpyError}
    />)
    expect(wrapper).toMatchSnapshot()
})

test('MapImportToStandardFields calls confirmImport when button is clicked', () => {
    const wrapper = shallow(<MapImportToStandardFields
        standardFields={standardFields}
        confirmImport={onClickSpyOk}
        importRowIdentifierStandardFieldName="Address"
        mapImportToStandardFieldsERROR={false}
        setMapImportToStandardFieldsERROR={onClickSpyError}
    />)
    wrapper.find('button').simulate('click', standardFields)
    expect(onClickSpyOk).toHaveBeenLastCalledWith(standardFields)
})

test('MapImportToStandardFields calls setMapImportToStandardFieldsERROR when ' +
     'standardFieldSet does not have any importRowIdentifier in it and the button is clicked', () => {
    const wrapper = shallow(<MapImportToStandardFields 
        standardFields={standardFieldsNoImportRowIdentifier}
        confirmImport={onClickSpyOk}
        importRowIdentifierStandardFieldName="Address"
        setMapImportToStandardFieldsERROR={onClickSpyError}
        mapImportToStandardFieldsERROR={false}
    />)
    wrapper.find('button').simulate('click', standardFieldsNoImportRowIdentifier)
    expect(onClickSpyError).toHaveBeenLastCalledWith()
})

test('MapImportToStandardFields calls setMapImportToStandardFieldsERROR when standardField with importRowIdentifier' +
     'does not have an importedFieldName and the button is clicked', () => {
    const wrapper = shallow(<MapImportToStandardFields 
        standardFields={standardFieldsNoImportedFieldName}
        confirmImport={onClickSpyOk}
        importRowIdentifierStandardFieldName="Address"
        setMapImportToStandardFieldsERROR={onClickSpyError}
        mapImportToStandardFieldsERROR={false}
    />)
    wrapper.find('button').simulate('click', standardFieldsNoImportedFieldName)
    expect(onClickSpyError).toHaveBeenLastCalledWith()
})