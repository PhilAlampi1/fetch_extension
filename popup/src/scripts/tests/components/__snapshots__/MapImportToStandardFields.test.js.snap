// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapImportToStandardFields renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapImportToStandardFields
    setMapImportToStandardFieldsERROR={[Function]}
    standardFields={
      Array [
        Object {
          "importRowIdentifier": false,
          "importedFieldName": "IDX Address Mapg",
          "standardFieldDescription": "A field for mapping.",
          "standardFieldId": "1",
          "standardFieldName": "Mapping Field (IDX)",
        },
        Object {
          "importRowIdentifier": false,
          "importedFieldName": "Property Type",
          "standardFieldDescription": "The type of building or structure the property is.",
          "standardFieldId": "2",
          "standardFieldName": "Type of Property",
        },
        Object {
          "importRowIdentifier": true,
          "importedFieldName": "Street Address",
          "standardFieldDescription": "Address of property.",
          "standardFieldId": "3",
          "standardFieldName": "Address",
        },
      ]
    }
    updateImportFileMapping={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Your file was uploaded successfully, nice work!
        </p>,
        <p>
          We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.
        </p>,
        <div
          className="list-header"
        >
          <div>
            Import Field
          </div>
          <div>
            Standard Field
          </div>
          <div>
            Description
          </div>
        </div>,
        <div
          className="list-body"
        >
          <ImportMappingLineItem
            handleOnChange={[Function]}
            importedFieldName="IDX Address Mapg"
            standardFieldDescription="A field for mapping."
            standardFieldId="1"
            standardFieldName="Mapping Field (IDX)"
          />
          <ImportMappingLineItem
            handleOnChange={[Function]}
            importedFieldName="Property Type"
            standardFieldDescription="The type of building or structure the property is."
            standardFieldId="2"
            standardFieldName="Type of Property"
          />
          <ImportMappingLineItem
            handleOnChange={[Function]}
            importedFieldName="Street Address"
            standardFieldDescription="Address of property."
            standardFieldId="3"
            standardFieldName="Address"
          />
        </div>,
        undefined,
        <button
          onClick={[Function]}
        >
          Next
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Your file was uploaded successfully, nice work!",
        },
        "ref": null,
        "rendered": "Your file was uploaded successfully, nice work!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.",
        },
        "ref": null,
        "rendered": "We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Import Field
            </div>,
            <div>
              Standard Field
            </div>,
            <div>
              Description
            </div>,
          ],
          "className": "list-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Import Field",
            },
            "ref": null,
            "rendered": "Import Field",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Standard Field",
            },
            "ref": null,
            "rendered": "Standard Field",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Description",
            },
            "ref": null,
            "rendered": "Description",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="IDX Address Mapg"
              standardFieldDescription="A field for mapping."
              standardFieldId="1"
              standardFieldName="Mapping Field (IDX)"
            />,
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="Property Type"
              standardFieldDescription="The type of building or structure the property is."
              standardFieldId="2"
              standardFieldName="Type of Property"
            />,
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="Street Address"
              standardFieldDescription="Address of property."
              standardFieldId="3"
              standardFieldName="Address"
            />,
          ],
          "className": "list-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "handleOnChange": [Function],
              "importedFieldName": "IDX Address Mapg",
              "standardFieldDescription": "A field for mapping.",
              "standardFieldId": "1",
              "standardFieldName": "Mapping Field (IDX)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "handleOnChange": [Function],
              "importedFieldName": "Property Type",
              "standardFieldDescription": "The type of building or structure the property is.",
              "standardFieldId": "2",
              "standardFieldName": "Type of Property",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "handleOnChange": [Function],
              "importedFieldName": "Street Address",
              "standardFieldDescription": "Address of property.",
              "standardFieldId": "3",
              "standardFieldName": "Address",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      undefined,
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Your file was uploaded successfully, nice work!
          </p>,
          <p>
            We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.
          </p>,
          <div
            className="list-header"
          >
            <div>
              Import Field
            </div>
            <div>
              Standard Field
            </div>
            <div>
              Description
            </div>
          </div>,
          <div
            className="list-body"
          >
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="IDX Address Mapg"
              standardFieldDescription="A field for mapping."
              standardFieldId="1"
              standardFieldName="Mapping Field (IDX)"
            />
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="Property Type"
              standardFieldDescription="The type of building or structure the property is."
              standardFieldId="2"
              standardFieldName="Type of Property"
            />
            <ImportMappingLineItem
              handleOnChange={[Function]}
              importedFieldName="Street Address"
              standardFieldDescription="Address of property."
              standardFieldId="3"
              standardFieldName="Address"
            />
          </div>,
          undefined,
          <button
            onClick={[Function]}
          >
            Next
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Your file was uploaded successfully, nice work!",
          },
          "ref": null,
          "rendered": "Your file was uploaded successfully, nice work!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.",
          },
          "ref": null,
          "rendered": "We've taken a stab at mapping your import file to the Standard Fields we use in the system. However, it's unlikely we got this completely right. This is a very important step, so please check this over and make corrections where needed.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Import Field
              </div>,
              <div>
                Standard Field
              </div>,
              <div>
                Description
              </div>,
            ],
            "className": "list-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Import Field",
              },
              "ref": null,
              "rendered": "Import Field",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Standard Field",
              },
              "ref": null,
              "rendered": "Standard Field",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Description",
              },
              "ref": null,
              "rendered": "Description",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ImportMappingLineItem
                handleOnChange={[Function]}
                importedFieldName="IDX Address Mapg"
                standardFieldDescription="A field for mapping."
                standardFieldId="1"
                standardFieldName="Mapping Field (IDX)"
              />,
              <ImportMappingLineItem
                handleOnChange={[Function]}
                importedFieldName="Property Type"
                standardFieldDescription="The type of building or structure the property is."
                standardFieldId="2"
                standardFieldName="Type of Property"
              />,
              <ImportMappingLineItem
                handleOnChange={[Function]}
                importedFieldName="Street Address"
                standardFieldDescription="Address of property."
                standardFieldId="3"
                standardFieldName="Address"
              />,
            ],
            "className": "list-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "handleOnChange": [Function],
                "importedFieldName": "IDX Address Mapg",
                "standardFieldDescription": "A field for mapping.",
                "standardFieldId": "1",
                "standardFieldName": "Mapping Field (IDX)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "handleOnChange": [Function],
                "importedFieldName": "Property Type",
                "standardFieldDescription": "The type of building or structure the property is.",
                "standardFieldId": "2",
                "standardFieldName": "Type of Property",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "function",
              "props": Object {
                "handleOnChange": [Function],
                "importedFieldName": "Street Address",
                "standardFieldDescription": "Address of property.",
                "standardFieldId": "3",
                "standardFieldName": "Address",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        undefined,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
